### PR TITLE
added a delete button to the form in order to delete events

### PR DESCRIPTION
**Pull request by CamilingJS**

### CODE REVIEW

File: app/routes/events_.$eventId.tsx [hunk 1]
```diff
@@ -29,13 +29,27 @@ export async function loader({ params }: LoaderFunctionArgs) {
...
+  switch (intent) {
+    case 'joinEvent': {
+      await db.event.update({
+        where: { id: params.eventId },
+        data: { users: { connect: { id: userSession.userId } } },
+      });
+      return redirect(`/events/${params.eventId}`);
+    }
+    case 'deleteEvent': {
+      await db.event.delete({
```
Comment (eiffelwong1) [inline app/routes/events_.$eventId.tsx:44]: in the future, we should probably make the delete action, just get marked as deleted, but not actually removed from the database.

so when participants re-visit the page it wouldn't be a 404

**Review by eiffelwong1 — APPROVED** [review]
LGTM, :shipit:

**Review by andrelandgraf — APPROVED** [review]
Great PR!

**Review by tonydangblog — APPROVED** [review]
Looks good to me! The usage of "intent" is very clever!

**Pull request merged**